dizoo/classic_control/mountain_car/envs/mtcar_env.py

```diff
@@ -0,0 +1,132 @@
+from typing import Any, List, Union, Optional
+import gym
+import numpy as np
+from ding.envs import BaseEnv, BaseEnvTimestep
+from ding.torch_utils import to_ndarray, to_list
+from ding.utils import ENV_REGISTRY
+
+@ENV_REGISTRY.register('mountain_car')
+class MountainCarEnv(BaseEnv):
+
+    """
+    Implementation of DI-engine's version of the Mountain Car deterministic MDP. 
+
+    Important references that contributed to the creation of this env:
+    > Source code of OpenAI's mountain car gym : https://is.gd/y1FkMT
+    > Gym documentation of mountain car : https://is.gd/29S0dt
+    > Based off DI-engine existing implementation of cartpole_env.py
+    > DI-engine's env creation conventions : https://is.gd/ZHLISj
+
+    Only __init__ , step, seed and reset are mandatory & impt.
+    The other methods are generally for convenience.
+    """
+
+    def __init__(self, cfg: dict = {}) -> None:
+        self._cfg = cfg
+        self._init_flag = False
+        self._replay_path = None
+
+        # Following specifications from https://is.gd/29S0dt
+        self._observation_space = gym.spaces.Box(
+            low=np.array([-1.2, -0.07]),
+            high=np.array([0.6, 0.07]),
+            shape=(2, ),
+            dtype=np.float32
+        )
+        self._action_space = gym.spaces.Discrete(3, start=0)
+        self._reward_space = gym.spaces.Box(low=-1, high=0.0, shape=(1, ), dtype=np.float32)
+
+    def seed(self, seed: int, dynamic_seed: bool = True) -> None:
+        self._seed = seed
+        self._dynamic_seed = dynamic_seed
+        np.random.seed(self._seed)
+
+    def reset(self) -> np.ndarray:
+        # Instantiate environment if not already done so
+        if not self._init_flag:
+            self._env = gym.make('MountainCar-v0')
+        self._init_flag = True
```

Comment:
this line need to be in above if-branch

Reply:
Ok. Edited this with 9ca9009e96cc94303d78efe6756b20891c34aa8a